lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,395 @@
+                             $(e.target).prev('.final-confirmation-btn');
+                         nextButton.focus();
+                     } else if (e.key === 'Escape') {
+                         openButton.popover('hide');
+                         openButton.focus();
+                     }
+                 },
+
+                 formatCourseHomeUrl: function(sessionKey) {
+                     /*
+                     Takes the base course home URL and updates it with the new session id, leveraging the
+                     the fact that all course keys contain a '+' symbol.
+                     */
+                     var oldSessionKey = this.courseHomeUrl.split('/')
+                         .filter(
+                             function(urlParam) {
+                                 return urlParam.indexOf('+') > 0;
+                             }
+                         )[0];
+                     return this.courseHomeUrl.replace(oldSessionKey, sessionKey);
+                 },
+
+                 formatDates: function(sessionData) {
+                     /*
+                     Takes a data object containing the upcoming available sessions for an entitlement and returns
+                     the object with a session_dates attribute representing a formatted date string that highlights
+                     the start and end dates of the particular session.
+                      */
+                     var formattedSessionData = sessionData,
+                         startDate,
+                         endDate,
+                         dateFormat;
+                     moment.locale(window.navigator.userLanguage || window.navigator.language);
+                     dateFormat = moment.localeData().longDateFormat('L').indexOf('DD') >
+                         moment.localeData().longDateFormat('L').indexOf('MM') ? 'MMMM D, YYYY' : 'D MMMM, YYYY';
+
+                     return _.map(formattedSessionData, function(session) {
+                         var formattedSession = session;
+                         startDate = formattedSession.session_start ?
+                             moment((new Date(formattedSession.session_start))).format(dateFormat) : null;
+                         endDate = formattedSession.session_end ?
+                             moment((new Date(formattedSession.session_end))).format(dateFormat) : null;
+                         formattedSession.session_dates = this.courseCardModel.formatDateString({
+                             start_date: startDate,
+                             end_date: endDate,
+                             pacing_type: formattedSession.pacing_type
+                         });
+                         formattedSession.enrollment_end = formattedSession.enrollment_end ?
+                             moment((new Date(formattedSession.enrollment_end))).format(dateFormat) : null;
```

Comment:
This call is repeated a lot, maybe it could be pulled out into a helper method and DRYd?

Reply:
Added helper